manifests/plugin/ansible.pp

```diff
@@ -16,7 +16,7 @@
   }
 
   pulpcore::plugin { 'ansible':
-    config        => "ANSIBLE_API_HOSTNAME = \"${pulpcore::servername}\"\nANSIBLE_CONTENT_HOSTNAME = \"${pulpcore::servername}\"",
+    config        => "ANSIBLE_API_HOSTNAME = \"${pulpcore::servername}\"\nANSIBLE_CONTENT_HOSTNAME = \"https://${pulpcore::servername}/pulp/content/\"",
```

Comment:
I don't like constructing a URL again and would prefer to reuse.

We have this here:
https://github.com/theforeman/puppet-pulpcore/blob/22db71ca2c220f52f6651e93d3c60bf19ef0e319/manifests/apache.pp#L19

What we should do is construct the URL. Technically speaking it should also take `$https_port` into account and fail if `https` is turned off. So roughly:

```puppet
if ! $pulpcore::apache_https_vhost {
  # TODO: or can it be served over HTTP?
  fail('HTTPS must be turned on for Ansible content')
} elsif $pulpcore::apache::https_port != 443 {
  # Technically we can't determine this when this module doesn't manage the vhost
  # It's up to the user to keep them in sync
  $external_content_url = "https://${pulpcore::servername}:${pulpcore::apache::https_port}${pulpcore::apache::content_path}"
} else {
  $external_content_url = "https://${pulpcore::servername}${pulpcore::apache::content_path}"
}
```

We're not so complete here:

https://github.com/theforeman/puppet-foreman_proxy_content/blob/fb401944723c5b93941aa98d16f377bded5627c0/manifests/init.pp#L299

It may be an idea to add a `$external_content_url` in `apache.pp` so that it can be used in puppet-foreman_proxy_content too.

Reply:
> It may be an idea to add a `$external_content_url` in `apache.pp` so that it can be used in puppet-foreman_proxy_content too.

I like this idea, and I think it's best that we tackle the change in apache.pp it in a separate PR.

We could fix the immediate issue here, then improve the cross-module implementation details in a PR that is "Refs #32766 - ..." (really two separate PRs, since we'd also want to update puppet-FPC at that point)